### PR TITLE
feat(sqla-repo): health_check generates valid statement for oracle.

### PR DESCRIPTION
**Pull request by cofin**
### Pull Request Checklist

This PR modifies the repo health check URL so that it generates a valid query for the Oracle dialect.

[//]: # "By submitting this issue, you agree to:"
[//]: # "- follow Litestar's [Code of Conduct](https://github.com/litestar-org/.github/blob/main/CODE_OF_CONDUCT.md)"
[//]: # "- follow Litestar's [Contribution Guidelines](https://litestar.dev/community/contribution-guide)"

- [ ] New code has 100% test coverage
- [ ] (If applicable) The prose documentation has been updated to reflect the changes introduced by this PR
- [ ] (If applicable) The reference documentation has been updated to reflect the changes introduced by this PR

### Description
[//]: # "Please describe your pull request for new release changelog purposes"
[//]: # "Example: 'This pr adds the capability to use server-sent events, and documents it in the usage docs'"

-

### Close Issue(s)
[//]: # "Please add in issue numbers this pull request will close, if applicable"
[//]: # "Examples: Fixes #4321 or Closes #1234"

-

### CODE REVIEW

File: litestar/contrib/sqlalchemy/repository/_sync.py [hunk 1]
```diff
@@ -715,8 +715,9 @@ def check_health(cls, session: Session) -> bool:
         Returns:
             `True` if healthy.
         """
+        sql = "SELECT 1 FROM DUAL" if session.bind and session.bind.dialect.name == "oracle" else "SELECT 1"
```
Comment (sobolevn) [inline litestar/contrib/sqlalchemy/repository/_sync.py:718]: Let's please refactor this into a separate function.

Comment (cofin) [reply litestar/contrib/sqlalchemy/repository/_sync.py]: @sobolevn Thanks for the review.  Take a look and see if the latest change is what you had in mind.

File: litestar/contrib/sqlalchemy/repository/_async.py [hunk 2]
```diff
@@ -713,10 +713,17 @@ async def check_health(cls, session: AsyncSession) -> bool:
         Returns:
             `True` if healthy.
```
Comment (JacobCoffee) [inline litestar/contrib/sqlalchemy/repository/_async.py:714]:
```suggestion
            ``True`` if healthy.
```
?

File: litestar/contrib/sqlalchemy/repository/_sync.py [hunk 3]
```diff
@@ -715,10 +715,17 @@ def check_health(cls, session: Session) -> bool:
         Returns:
             `True` if healthy.
```
Comment (JacobCoffee) [inline litestar/contrib/sqlalchemy/repository/_sync.py:716]:
```suggestion
            ``True`` if healthy.
```
?

**Review by JacobCoffee — APPROVED** [review]

**Review by provinzkraut — APPROVED** [review]

**Pull request merged**